### PR TITLE
Layers UI

### LINKED ISSUE #256: Add Layer UI in the visualizer
### Description of the task

Add UI elements to represent an array of layers in the visualizer page of corner kick

### Outline the task in more detail

We will be using the work from #248 to define what needs to be represented in the layer UI element.
This UI will appear in the sidebar and should take the form of a panel

### Acceptance criteria

- [ ] Represent an array of layer in the form of UI elements
- [ ] Represent each field value for a layer in the UI as defined in #248

### PR DESCRIPTION
**Pull request by qgolsteyn**
### Description

Adds a Layer Panel UI and additional Storybook elements to correctly display the UI changes. Changes have been reflected in the Storybook.

### Testing Done

Visual testing in Storybook

### Resolved Issues

Closes #256 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [X] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [X] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [X] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/corner_kick/src/components/LayersPanel/__stories__/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+const stories = storiesOf('Layers Panel', module);
+
+/**
+ * Story describing a LayerPanel with some layers defined
+ */
+stories.add('with layers', () => (
+    <StorybookStage width="300px">
+        <LayersPanel
+            layers={[
```
Comment (MathewMacDougall) [inline src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:19]: Just to clarify, these layers will later be dynamically created from the `Layer.msg` ROS messages? We won't need to hardcode layers right?

**Review by MathewMacDougall — COMMENTED** [review]
One quick clarification but otherwise LGTM

File: src/corner_kick/src/components/LayersPanel/__stories__/index.tsx (hunk 1 above)
Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/__stories__/index.tsx:19]: Yup. This is just a visual test to make sure the UI is working correctly.

You can see how it looks here: https://deploy-preview-271--adoring-mestorf-e664ee.netlify.com/?selectedKind=Layers%20Panel&selectedStory=with%20layers&full=0&addons=1&stories=1&panelRight=0

**Review by MathewMacDougall — DISMISSED** [review]
LGTM :+1:

File: src/corner_kick/src/components/LayersPanel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
+/*
+ * This file defines the UI to control the various layers of the visualizer
```
Comment (garethellis0) [inline src/corner_kick/src/components/LayersPanel/index.tsx]: Are we referring to the layers that we draw over the "field" section of the visualizer here? Or over the entire visualizer?

Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: We referring to the UI that controls layer order and visibility. It will go in the sidebar.

Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: Oh lol, referring to every layer of the visualizer, field is just one of them

**Comment by qgolsteyn** [comment]
Pushed another commit to trigger build and reduce the number of CI checks netlify is conducting

**Review by MathewMacDougall — DISMISSED** [review]

File: src/corner_kick/src/components/LayersPanel/index.tsx (hunk 2 above)
Comment (garethellis0) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: Ah, could we make that a bit more clear here? My confusion was coming from the fact that we're using the word "layer" to refer to the layers of shapes that we draw over the field. Would be good to indicate that this is not the same thing.

**Comment by garethellis0** [comment]
I re-triggered the travis build (which, just as a note, you can do without re-pushing).

File: src/corner_kick/src/components/LayersPanel/index.tsx (hunk 2 above)
Comment (MathewMacDougall) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: Also to clarify, the "Layers" here still appear on the "Field Area" of the Visualizer right? We can't draw random shapes that appear magically in other places, right?

Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: Okay... Hum I'm not sure what you are talking about. The UI described in this PR is meant to control the z-ordering and visibility of the layers in the visualizer. I suppose by "field area" you are talking about the area where the content (ie. shapes) of the layers are drawn. That "field area" needs additional information (ie. in which order should layers be drawn; is a layer visible or not) which the user can control through the UI in this PR

Comment (MathewMacDougall) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: Sorry, I think "Canvas" is a better term. The area where the AI can request shapes to be drawn is separate from the console on the bottom, and control panel on the left?

Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: I suppose you could say that. More specifically however is the AI can send shapes to be drawn to the visualizer, and the visualizer can choose where to draw those shapes and add information it deems necessary for visualization. This information can come from the user via additional controls in the Sidebar or Console area (in fact — this is what this PR provides, a control to control the visibility and z-ordering of layers, which is something that the AI does not provide).

TLDR: we need planning documents.

Comment (MathewMacDougall) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: We're good

Comment (garethellis0) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: This makes sense, could we rephrase the comment to make the above more clear? Perhaps something like: "This file defines the UI to control the layers in the field area of the visualizer"?

Comment (MathewMacDougall) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: "field area" isn't the correct term to use here (and was the cause of the confusion above). If anything, I think all we need here is to specify the layers are on the "canvas".

File: src/corner_kick/src/components/LayersPanel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
+/*
+ * This file defines the UI to control the layers of the visualizer
```
Comment (qgolsteyn) [inline src/corner_kick/src/components/LayersPanel/index.tsx]:
``` suggestion
 * This file defines the UI to control the layers in the Canvas
```

File: src/corner_kick/src/components/LayersPanel/index.tsx (hunk 2 above)
Comment (qgolsteyn) [reply src/corner_kick/src/components/LayersPanel/index.tsx]: Update file description to "This file defines the UI to control the layers in the Canvas"

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:
